torchdata/datapipes/iter/util/unzipper.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import types
+from typing import Optional, Sequence, TypeVar
+
+from torch.utils.data.datapipes.iter.combining import _ChildDataPipe, _ForkerIterDataPipe
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+T = TypeVar("T")
+
+
+@functional_datapipe("unzip")
+class UnZipperIterDataPipe(IterDataPipe[T]):
+    r"""
+    Takes in a DataPipe of Sequences, unpacks each Sequence, and return the elements in separate DataPipes
+    based on their position in the Sequence.
+
+    Note:
+        Each sequence within the DataPipe should have the same length, optionally specified by
+        the input argument `sequence_length`.
+
+    Args:
+        source_datapipe: Iterable DataPipe with sequences of data
+        sequence_length: Length of the sequence within the source_datapipe. All elements should have the same length.
+            If `None`, it is inferred from the first sequence in the `source_datapipe`.
+        buffer_size: this restricts how far ahead the leading child DataPipe can read relative
+            to the slowest child DataPipe. Use -1 for the unlimited buffer.
+    """
+
+    def __new__(
+        cls, source_datapipe: IterDataPipe[Sequence[T]], sequence_length: Optional[int] = None, buffer_size: int = 1000
```

Comment:
We may want to add an argument for users to specify the number of instances then we don't need to fetch the first data from prior datapipe. (Lazy loading)
And, another argument can also be introduced to drop specific column(s) from input.

Reply:
`sequence_length` does the first thing that you describe. It is currently optional, should I make it required?

I can add the second thing.

Reply:
Let's make it required. Otherwise, either serialization won't work or we need to trigger iteration of prior DataPipe twice.

Reply:
Done